pkgs/servers/nosql/arangodb/default.nix

```diff
@@ -1,68 +1,81 @@
-{ stdenv, lib, fetchFromGitHub, openssl, zlib, cmake, python2, perl, snappy, lzo, which, catch2, catch }:
+{
+  # gcc 11.2 suggested on 3.10.0.
+  # gcc 11.3.0 unsupported yet, investigate gcc support when upgrading
+  # See https://github.com/arangodb/arangodb/issues/17454
+  gcc10Stdenv
+, git
+, lib
+, fetchFromGitHub
+, openssl
+, zlib
+, cmake
+, python3
+, perl
+, snappy
+, lzo
+, which
+, targetArchitecture ? null
+, asmOptimizations ? gcc10Stdenv.targetPlatform.isx86
+}:
 
 let
-  common = { version, sha256 }: stdenv.mkDerivation {
-    pname = "arangodb";
-    inherit version;
+  defaultTargetArchitecture =
+    if gcc10Stdenv.targetPlatform.isx86
+    then "haswell"
+    else "core";
 
-    src = fetchFromGitHub {
-      repo = "arangodb";
-      owner = "arangodb";
-      rev = "v${version}";
-      inherit sha256;
-    };
+  targetArch =
+    if isNull targetArchitecture
+    then defaultTargetArchitecture
+    else targetArchitecture;
+in
 
-    nativeBuildInputs = [ cmake python2 perl which ];
-    buildInputs = [ openssl zlib snappy lzo ];
+gcc10Stdenv.mkDerivation rec {
+  pname = "arangodb";
+  version = "3.10.0";
 
-    # prevent failing with "cmake-3.13.4/nix-support/setup-hook: line 10: ./3rdParty/rocksdb/RocksDBConfig.cmake.in: No such file or directory"
-    dontFixCmake       =                     lib.versionAtLeast version "3.5";
-    NIX_CFLAGS_COMPILE = lib.optionalString (lib.versionAtLeast version "3.5") "-Wno-error";
-    preConfigure       = lib.optionalString (lib.versionAtLeast version "3.5") "patchShebangs utils";
+  src = fetchFromGitHub {
+    repo = "arangodb";
+    owner = "arangodb";
+    rev = "v${version}";
+    sha256 = "0vjdiarfnvpfl4hnqgr7jigxgq3b3zhx88n8liv1zqa1nlvykfrb";
+    fetchSubmodules = true;
+  };
 
-    postPatch = ''
-      sed -ie 's!/bin/echo!echo!' 3rdParty/V8/v*/gypfiles/*.gypi
+  nativeBuildInputs = [ cmake git perl python3 which ];
 
-      # with nixpkgs, it has no sense to check for a version update
-      substituteInPlace js/client/client.js --replace "require('@arangodb').checkAvailableVersions();" ""
-      substituteInPlace js/server/server.js --replace "require('@arangodb').checkAvailableVersions();" ""
+  buildInputs = [ openssl zlib snappy lzo ];
 
-      ${if (lib.versionOlder version "3.4") then ''
-        cp ${catch}/include/catch/catch.hpp 3rdParty/catch/catch.hpp
-      '' else if (lib.versionOlder version "3.5") then ''
-        cp ${catch2}/include/catch2/catch.hpp 3rdParty/catch/catch.hpp
-      '' else ''
-        (cd 3rdParty/boost/1.69.0 && patch -p1 < ${../../../development/libraries/boost/pthread-stack-min-fix.patch})
-      ''}
-    '';
+  # prevent failing with "cmake-3.13.4/nix-support/setup-hook: line 10: ./3rdParty/rocksdb/RocksDBConfig.cmake.in: No such file or directory"
+  dontFixCmake = true;
+  NIX_CFLAGS_COMPILE = "-Wno-error";
+  preConfigure = "patchShebangs utils";
```

Comment:
This should really be a multiline string.

Reply:
Thank you! https://github.com/NixOS/nixpkgs/pull/200210